### ISSUE TITLE
KVM: -drive if=scsi is deprecated on PC machine types

### ISSUE THREAD

**Issue by apoikos** [ganeti/ganeti#1341]
See https://github.com/qemu/qemu/commit/f778a82f0c179634892ea716970d4d35264dc841. This is yet another reason to address #762 at some point.

**Comment by paravoid** [comment]
qemu/qemu@1454509726719e0933c800fad00d6999752688ea removed this entirely, breaking VMs with disk_type SCSI:
```
Could not start instance 'vm.example.org': Hypervisor error: Failed to start instance vm.example.org: exited with exit code 1 (qemu-system-x86_64: -usbdevice tablet: '-usbdevice' is deprecated, please use '-device usb-...' instead
qemu-system-x86_64: -drive file=/var/run/ganeti/instance-disks/vm.example.org:0,format=raw,if=scsi: machine type does not support if=scsi,bus=0,unit=0
```

Also note that this commit is part of qemu v2.12, and is affecting e.g. Debian buster.

**Comment by paravoid** [comment]
Also unrelated to the breakage: the first commit mentions: _lsi53c895a is thoroughly obsolete (PCI Ultra2 SCSI, ca. 2000), and currently has no maintainer in QEMU_. Ganeti defaults `scsi_controller_type` to `lsi` that should probably be changed to `megasas`.

**Comment by benjamir** [comment]
@paravoid: agree, related to "lsi" vs "megasas" were Oracle urges to use "megasas":
"KVM guests with LSI MegaRAID SAS ISCSI controller limited to 7 virtual disks"
 https://docs.oracle.com/cd/E52668_01/E93593/html/ol7-bug-27681238.html
(Documented for Oracle 7.5, but I don't see a reason that it doesn't apply for all current distributions:)
